build.gradle

```diff
@@ -0,0 +1,284 @@
+      }
+      dependency 'com.github.kevinsawicki:http-request:5.4.1'
+      dependency 'org.codehaus.sonar.runner:sonar-runner-api:2.4'
+      dependency 'org.freemarker:freemarker:2.3.20'
+      dependency "com.google.protobuf:protobuf-java:${protobufVersion}"
+      dependency 'net.jpountz.lz4:lz4:1.3.0'
+      dependency 'org.subethamail:subethasmtp:3.1.7'
+      dependencySet(group: 'org.apache.tomcat.embed', version: '8.5.23') {
+        entry 'tomcat-embed-core'
+        entry('tomcat-embed-jasper') {
+          exclude 'org.eclipse.jdt.core.compiler:ecj'
+        }
+      }
+      dependencySet(group: 'com.squareup.okhttp3', version: '3.7.0') {
+        entry 'okhttp'
+        entry 'mockwebserver'
+      }
+      // TODO provided
+      dependency 'xml-apis:xml-apis:1.4.01'
+      dependency 'org.sonarsource.orchestrator:sonar-orchestrator:3.16.0.1420'
+    }
+  }
+
+  apply plugin: 'java'
+  sourceCompatibility = 1.8
+  tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+  }
+
+  def tempDir = file("$buildDir/temp")
+  test.doFirst {
+    // TODO note that due to this any change in this file (even addition of blank line) is considered as modification of input for tests
+    delete tempDir
+    tempDir.mkdirs()
+  }
+  test {
+    systemProperty 'java.io.tmpdir', tempDir
+    systemProperty 'java.awt.headless', true 
+    maxHeapSize = "512m"
+    jacoco.enabled = project.hasProperty('jacocoEnabled')
+    testLogging {
+      exceptionFormat 'full'
+      events "skipped", "failed"
+    }
+  }
+
+  apply plugin: 'com.google.protobuf'
+  sourceSets.main.proto.srcDir 'src/main/protobuf'  // in addition to the default 'src/main/proto'
+  sourceSets.test.proto.srcDir 'src/test/protobuf'  // in addition to the default 'src/test/proto'
+  protobuf {
+    protoc {
+      artifact = "com.google.protobuf:protoc:${protobufVersion}"
+    }
+  }
+
+  if (release) {
+    jar {
+      manifest {
+        attributes(
+          'Version': "${version}",
+          'Implementation-Build': 'git rev-parse HEAD'.execute().text.trim(),
+          'Build-Time': new Date().format("yyyy-MM-dd'T'HH:mm:ssZ")
+        )
+      }
+    }
+  }
+
+  apply plugin: 'idea'
+  idea {
+    module {
+      sourceDirs += file("${protobuf.generatedFilesBaseDir}/main/java")
+      testSourceDirs += file("${protobuf.generatedFilesBaseDir}/test/java")
+    }
+  }
+}
+
+configurations {
+  archives
+}
+
+apply plugin: 'maven-publish'
+// TODO requires review
+artifactory {
+  clientConfig.setIncludeEnvVars(true)
+  clientConfig.setEnvVarsExcludePatterns('*password*,*PASSWORD*,*secret*,*MAVEN_CMD_LINE_ARGS*,sun.java.command,*token*,*TOKEN*,*LOGIN*,*login*')
+  contextUrl = System.getenv('ARTIFACTORY_URL')
+  publish {
+    repository {
+      repoKey = System.getenv('ARTIFACTORY_DEPLOY_REPO')
+      username = System.getenv('ARTIFACTORY_DEPLOY_USERNAME')
+      password = System.getenv('ARTIFACTORY_DEPLOY_PASSWORD')
+    }
+    defaults {
+      properties = [
```

Comment:
the Travis specific env variables are not required anymore. Generic variables are defined in travis.sh.

Reply:
Good catch!

Reply:
Done